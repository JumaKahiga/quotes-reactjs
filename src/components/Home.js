import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { dayQuote } from '../actions/quotesActions';


class DayQuote extends Component {
	componentWillMount() {
		this.props.dayQuote();
	}

	componentWillReceiveProps(nextProps) {
		if(nextProps.newQuote){
			this.props.quotes.unshift(nextProps.newQuote);
		}

	}

	render() {
		const quoteItems = this.props.myQuote.map(quote =>(
			<div className="center" key>{ quote.id }
				<p>{ quote.quote }</p>
				<h6>Posted { quote.posted } ago by { quote.author }</h6>
			</div>
			))
		return(
		<div>
			<h3 className="center">Quote of the Day</h3> <hr />
			{ quoteItems }
		</div>
		)
	}
}

DayQuote.propTypes = {
	dayQuote: PropTypes.func.isRequired,
	quotes: PropTypes.array.isRequired,
	newQuote: PropTypes.object
};

const mapStateToProps = state => ({
	quotes: state.quotes.items,
	newQuote: state.quotes.item,
	myQuote: state.quotes.select_item
});

export default connect(mapStateToProps, { dayQuote })(DayQuote);