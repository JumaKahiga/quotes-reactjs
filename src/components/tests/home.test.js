import React from 'react';
import { shallow } from 'enzyme';
import { DayQuote } from '../Home';


function setup() {
  const props = {
    dayQuote: jest.fn(),
    myQuote: [{id: 2}],
    quotes: []
  };

  const enzymeWrapper = shallow(<DayQuote {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  describe('DayQuote', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();

      enzymeWrapper.setProps({
        dayQuote: jest.fn(),
        myQuote: [{id: 2}],
        newQuote: {},
        quotes: [],
      })

      expect(enzymeWrapper.find('h3').hasClass('center')).toBe(true);

      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});