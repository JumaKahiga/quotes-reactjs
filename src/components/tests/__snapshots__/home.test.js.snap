// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components DayQuote should render self and subcomponents 1`] = `ShallowWrapper {}`;
